src/main.rs

```diff
@@ -90,6 +90,52 @@ async fn main() -> Result<(), ()> {
         }
     }
 
+    //    On startup error out if the following checks don't pass:
+    // - That GetVersion has peersrpc, signerrpc and dev build tags.
+    // - That the version is > v0.16.0-beta
+    let version_info = client
+        .versioner()
+        .get_version(VersionRequest {})
+        .await
+        .expect("failed to get version")
+        .into_inner();
+    let version = &version_info.version;
+    let tags = &version_info.build_tags;
+
+    let min_version = String::from("0.16.0-beta");
+
+    let mut needs_peersrpc = true;
+    let mut needs_signerrpc = true;
+    let mut needs_dev = true;
+
+    for string in tags {
+        match string.as_str() {
+            "peersrpc" => needs_peersrpc = false,
+            "signerrpc" => needs_signerrpc = false,
+            "dev" => needs_dev = false,
+            _ => {}
+        }
+    }
+
+    if version < &min_version {
```

Comment:
Would prefer to do numerical comparison of major / minor / patch (rather than string comparison), while they technically achieve the same thing, using numerical values seems more sane.